designer/server/src/routes/forms/create.js

```diff
@@ -230,31 +222,40 @@ export default [
     async handler(request, h) {
       const { auth, payload, yar } = request
       const author = getAuthor(auth.credentials)
+      const token = auth.credentials.token
 
       // Update form metadata
-      const metadata = yar.set(sessionNames.create, {
+      yar.set(sessionNames.create, {
         ...yar.get(sessionNames.create),
         teamName: payload.teamName,
         teamEmail: payload.teamEmail
       })
 
       try {
-        const slug = slugify(metadata.title)
-
         // Create the form
-        await forms.create(payload, author)
+        const result = await forms.create(payload, author, token)
 
         // Clear form metadata
         yar.clear(sessionNames.create)
 
         /**
          * Redirect POST to GET without resubmit on back button
          */
-        return h.redirect(`/library/${slug}`).code(303)
-      } catch (cause) {
+        return h.redirect(`/library/${result?.slug}`).code(303)
+      } catch (err) {
+        if (Boom.isBoom(err) && err.data?.error === 'FormAlreadyExistsError') {
```

Comment:
Got some squiggles here, but I'm happy to help narrow this down

Glad it wasn't too much work